### PR TITLE
Ensure products are created in Admin scope first

### PR DESCRIPTION
**Pull request by convenient**
I have seen issues where
- products are created in a store specific scope
- updated in the admin scope
- Some of the values in the store specific scope incorrectly have "Use Default Value" set against them

This exception will ensure that products are created and fully defined in the admin scope prior to any store specific updates.

### CODE REVIEW

**Comment by convenient** [comment]
Reviewers, contact me directly if you want the ticket number or further details.

**Review by darryla — APPROVED** [review]
This is the least terrible option.

**Review by nei — APPROVED** [review]

**Review by thisisandrew — APPROVED** [review]
least terrible 🥇

File: Model/ProductRepository.php [hunk 1]
```diff
@@ -1,25 +1,41 @@
...
     ) {
         $this->vanillaRepository = $vanillaRepository;
         $this->productDataMapper = $productDataMapper;
+        $this->storeManager = $storeManager;
     }
 
     public function save(ProductInterface $product, $saveOptions = false)
     {
+        if ($this->storeManager->getStore()->getCode() !== Store::ADMIN_CODE) {
+            if (!$product->getIdBySku($product->getSku())) {
```
Comment (joshdifabio) [inline Model/ProductRepository.php]: As far as I know, `getIdBySku()` is not defined on `ProductInterface`, which means this is relying on a specific concretion being used (`Mage\Catalog\Model\Product`) and therefore violating the API. Perhaps you could inject the product resource model instead and call `getIdBySku()` on that directly.

Comment (convenient) [reply Model/ProductRepository.php]: correct

Thanks @joshdifabio, rebased.

**Pull request merged**